addons/medical/functions/fnc_treatment.sqf

```diff
@@ -201,6 +201,12 @@ _treatmentTime = if (isNumber (_config >> "treatmentTime")) then {
 };
 
 // Start treatment
+if (_caller != ACE_player) exitWith {
+    // It's an AI that is doing the treatment
+    sleep _treatmentTime;
+    [[_caller, _target, _selectionName, _className, _items, _usersOfItems]] call DFUNC(treatment_success);
```

Comment:
You already quoted one;
```sqf
 alive _player
```` 
And `canInteractWith` is a very important one as well. Also we use the onProgress callback for some treatment functions. In short, you would need to mimic the progress bar without the UI elements - so it's a PFH that checks all relevant conditions + does the onProgress callback. And when stopped, it would trigger failed.